arangod/IResearch/IResearchDataStore.cpp

```diff
@@ -550,53 +589,152 @@ void ConsolidationTask::operator()() {
 
 IResearchDataStore::IResearchDataStore(IndexId iid,
                                        LogicalCollection& collection)
-    : _engine(nullptr),
-      _asyncFeature(
+    : _asyncFeature(
           &collection.vocbase().server().getFeature<IResearchFeature>()),
       // mark as data store not initialized
       _asyncSelf(std::make_shared<AsyncLinkHandle>(nullptr)),
       _collection(collection),
       _error(DataStoreError::kNoError),
       _maintenanceState(std::make_shared<MaintenanceState>()),
-      _id(iid),
-      _lastCommittedTick(0),
-      _cleanupIntervalCount{0},
-      _numFailedCommits{nullptr},
-      _numFailedCleanups{nullptr},
-      _numFailedConsolidations{nullptr},
-      _commitTimeNum{0},
-      _avgCommitTimeMs{nullptr},
-      _cleanupTimeNum{0},
-      _avgCleanupTimeMs{nullptr},
-      _consolidationTimeNum{0},
-      _avgConsolidationTimeMs{nullptr},
-      _metricStats{nullptr} {
+      _id(iid) {
   // initialize transaction callback
-  _trxCallback = [this](transaction::Methods& trx,
-                        transaction::Status status) -> void {
+
+  _trxBeforeFinish = [this](transaction::Methods& trx) {
```

Comment:
NIT
Technically this is "PreCommit"  as transaction may be finished with Abort an this callback would not be called in such case.
```suggestion
  _trxPreCommit = [this](transaction::Methods& trx) {
```

Reply:
Hmm, yes, maybe I need to change it